idaes/generic_models/unit_models/heat_exchanger.py

```diff
@@ -330,10 +344,16 @@ def build(self):
         # compatability with older models.  Using add_object_reference keeps
         # these from showing up when you iterate through pyomo compoents in a
         # model, so only the user specified control volume names are "seen"
-        if not hasattr(self, "side_1"):
-            add_object_reference(self, "side_1", hot_side)
-        if not hasattr(self, "side_2"):
-            add_object_reference(self, "side_2", cold_side)
+        if config.side_1_is_hot:
```

Comment:
I wonder if we should have a deprecation warning here too (to try an push things towards using the standard names). I.e. let the user provide aliases, but also force them to have the standard names in the background. (and log a deprecation warning if they don't).

Reply:
I don't know.  I like the idea, but then the boiler heat exchanger would raise deprecation warning that the user couldn't fix.